ide-common/src/main/java/org/digma/intellij/plugin/insights/view/DurationBreakdownHelper.java

```diff
@@ -0,0 +1,39 @@
+package org.digma.intellij.plugin.insights.view;
+
+import com.intellij.openapi.project.Project;
+import org.digma.intellij.plugin.document.CodeObjectsUtil;
+import org.digma.intellij.plugin.model.rest.insights.SpanDurationBreakdown;
+import org.digma.intellij.plugin.psi.LanguageService;
+import org.digma.intellij.plugin.ui.model.listview.ListViewItem;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DurationBreakdownHelper {
+
+    private DurationBreakdownHelper() {
+    }
+
+    public static void findWorkspaceUrisForDurationBreakdowns(Project project, ListViewItem<?> theListView, @NotNull List<SpanDurationBreakdown> durationBreakdowns, @Nullable String methodCodeObjectId) {
```

Comment:
this class looks very similar to `SlowestSpansHelper`,
can you try to make reuse of code, instead of copy-paste?
maybe try to extract the ID's outside this method? and later call the method with span ids?

Reply:
great idea! thanks!